### PR TITLE
Jts habit details screen

### PR DESCRIPTION
**Pull request by johnschoeman**
|habit details|
|----|
|Simulator Screen Shot - iPhone 11 Pro Max - 2020-11-29 at 19 51 36|

### CODE REVIEW

File: src/HabitDetails/ProgressBar.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+const NUMBER_OF_DAYS = 21
+
+const ProgressBar: FunctionComponent<ProgressBarProps> = ({ habit }) => {
+  const [totalProgressBarWidth, setTotalProgressBarWidth] = useState<number>(0)
+  const days = habit.history.length
+  const daysCompleted = days > NUMBER_OF_DAYS ? NUMBER_OF_DAYS : days
+
+  const handleOnLayout = ({
```
Comment (devinjameson) [inline src/HabitDetails/ProgressBar.tsx:20]: Ahhhh this is much better. Should we update this on the home screen as well?

**Review by devinjameson — APPROVED** [review]
You're a designer bro. Agreed re: modals on modals. Not a good look. How about the detail screen shifts the screen left and has a native "Back" button type thing? And then the edit screen is a Modal™?

File: src/HabitDetails/ProgressBar.tsx (hunk 1 above)
Comment (johnschoeman) [reply src/HabitDetails/ProgressBar.tsx:20]: I think that'd be a good refactor! I also think having a percentage based calculation might be better, but not sure if itd be worth the effort.

**Comment by johnschoeman** [comment]
> Agreed re: modals on modals. Not a good look. How about the detail screen shifts the screen left and has a native "Back" button type thing? And then the edit screen is a Modal™?

Yeah that makes sense to me! Probably for a follow up commit though.

File: src/HabitDetails/ProgressBar.tsx (hunk 1 above)
Comment (devinjameson) [reply src/HabitDetails/ProgressBar.tsx:20]: So like X% done you mean?

Comment (johnschoeman) [reply src/HabitDetails/ProgressBar.tsx:20]: yup

**Pull request merged**